src/drivers/interrupt.c

```diff
@@ -27,6 +27,17 @@ DECLARE_SOF_UUID("irq", irq_uuid, 0x1862d39a, 0x3a84, 0x4d64,
 
 DECLARE_TR_CTX(irq_tr, SOF_UUID(irq_uuid), LOG_LEVEL_INFO);
 
+/* For i.MX with Zephyr make any second level
+ * interrupt handling go through interrupt-irqsteer.c
+ */
```

Comment:
If you have to re-spin this PR to address @marc-hb 's comment, could you also add to this comment, that because of this currently i.MX with Zephyr has to use both wrapper.c and this file which causes name collisions. That's why you need these defines. And below where you `#undef` for the first time, please add a comment saying, that that will now call the version from wrapper.c
@lgirdwood in fact this seems like a good candidate for an early native Zephyr API conversion: if we could switch over to native `z_soc_irq_*()` everywhere in the code with wrappers for XTOS builds, we wouldn't need this?

Reply:
Good point. @lyakh @iuliana-prodan what's the effort in switching over the APIs for Intel and NXP code ?

Reply:
@marc-hb the irq wrapping as discussed on call